lib/analyzer/applications/php.ts

```diff
@@ -22,13 +26,25 @@ export async function phpFilesToScannedProjects(
   const shouldIncludeDevDependencies = false;
 
   for (const pathPair of filePairs) {
-    const parserResult = await lockFileParser.buildDepTree(
-      filePathToContent[pathPair.lock],
-      filePathToContent[pathPair.manifest],
-      pathPair.manifest,
-      {},
-      shouldIncludeDevDependencies,
-    );
+    let parserResult: ComposerParserResponse | undefined;
+    try {
+      parserResult = buildDepTree(
+        filePathToContent[pathPair.lock],
+        filePathToContent[pathPair.manifest],
+        pathPair.manifest,
+        {},
+        shouldIncludeDevDependencies,
+      );
+    } catch (e) {
+      // This will skip parsing all files that error due to being malformed.
+      // If we do not do this, the entire scan will fail.
+      // Ideally, we'd like to log this, but logging does not exist in this library.
+      if (e instanceof InvalidUserInputError) {
```

Comment:
Could there be other `InvalidUserInputError` that we are skipping but don't want to.

Reply:
I spoke to Robert about this privately, and added the conclusion in the commit message.